### ISSUE TITLE
Should we add enum.equals(anotherEnum) to bug pattern?

### ISSUE THREAD

**Issue by pramoth** [google/error-prone#679]
I have sometime found a bug on **enum.equals(anotherEnum)**  because **anotherEnum** can be any object. So we lost type safety.
I think there is no valid use case for  **enum.equals(anotherEnum)**  .
**enum.equals(anotherEnum)** should replace with **enum == anotherEnum** to let  compiler check type for us.

**Comment by cushon** [comment]
There's already a check that should catch those errors: http://errorprone.info/bugpattern/EqualsIncompatibleType

Are you seeing examples where it doesn't detect an invalid use of `Object.equals` on an enum?

```java
class T {
  enum A {}
  enum B {}
  boolean f(A a, B b) {
    return a.equals(b);
  }
}
```

```
T.java:5: warning: [EqualsIncompatibleType] Calling equals on incompatible types A and B
    return a.equals(b);
                   ^
    (see http://errorprone.info/bugpattern/EqualsIncompatibleType)
1 warning
```

**Comment by pramoth** [comment]
it not detect this pattern ( method**isTypeEquals**)
```java
public class TestEnum {
    public enum TYPE {A, B}

    private TYPE type = TYPE.A;
 //detect if isTypeEquals(String type)
    public boolean isTypeEquals(Object type) {
        return this.type.equals(type);
    }
}

////////
TestEnum testEnum = new TestEnum();
testEnum.isTypeEquals("C")
```
It maybe stupid example but can happen.

**Comment by pramoth** [comment]
Thank you. It already.
